oem/ibm/libpldmresponder/file_io_type_dump.cpp

```diff
@@ -199,27 +199,79 @@ int DumpHandler::writeFromMemory(uint32_t, uint32_t length, uint64_t address,
     std::cout
         << "Start of DumpHandler::writeFromMemory transferFileDataToSocket"
         << std::endl;
-    return transferFileDataToSocket(DumpHandler::fd, length, address);
+
+    if (socketWriteStatus==Error)
```

Comment:
Suppose there is a case where 2 threads are running/waiting at a specific timeframe , when one thread hits error and we stop offloading the dump, how are we cleaning up the other thread?

Reply:
at a time only one thread allowed to run.
by any chance we hit case where two threads are there, as soon as first thread hit error, socket clean up and 2nd thread also hit error it will be cleaned up like first thread.
having critical section with mutex I don't think we hit this case

Reply:
There can be more than one thread at a time. But they will be waiting on the critical section for read/write operation. Yes, seems like they all can get cleaned up when the socketWriteStatus==Error happens. 

1. Who resets this socketWriteStatus to Free for the next dump offload?
2. At line 212 you will not get the socketInterface at the 2nd thread if its already cleaned up by the 1st thread. You will need to add a validity check before std::remove.
3. Does this PLDM_ERROR is sent to the phyp ? Will that not be a redundant message to phyp if its sent from all threads ?

Reply:
yeah we should avoid creating one more while one thread is running, otherwise we may end up issues you mentioned 
I will make have one thread at a time.